storage.go

```diff
@@ -2,35 +2,222 @@ package gdrive
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	o = s.newObject(false)
+	o.ID = s.getAbsPath(path)
+	o.Path = path
+	return o
+}
+
+func (s *Storage) createDir(ctx context.Context, name string, parents string) (fileId string, err error) {
+	dir := &drive.File{
+		Name:     name,
+		Parents:  []string{parents},
+		MimeType: "application/vnd.google-apps.folder",
+	}
+	f, err := s.service.Files.Create(dir).Context(ctx).Do()
+	if err != nil {
+		return "", err
+	}
+	return f.Id, nil
 }
 
 func (s *Storage) delete(ctx context.Context, path string, opt pairStorageDelete) (err error) {
-	panic("not implemented")
+	var fileId string
+	fileId, err = s.pathToId(ctx, path)
+	err = s.service.Files.Delete(fileId).Do()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get FileID via it's name and parents' fileID
+func (s *Storage) getFile(ctx context.Context, name string, parents string) ([]*drive.File, error) {
+	searchArg := fmt.Sprintf("name='%s' and parents='%s'", name, parents)
+	return s.rawListFiles(ctx, searchArg)
 }
 
 func (s *Storage) list(ctx context.Context, path string, opt pairStorageList) (oi *ObjectIterator, err error) {
-	panic("not implemented")
+	nextFn := func(ctx context.Context, page *ObjectPage) error {
+		fs, err := s.listFiles(ctx, path)
+		if err != nil {
+			return err
+		}
+		for _, f := range fs {
+			o := NewObject(s, true)
+			o.Path = f.Name
+
+			switch f.MimeType {
+			case "application/vnd.google-apps.folder":
+				o.Mode |= ModeDir
+			default:
+				o.Mode |= ModeRead
+			}
+			o.SetContentLength(f.Size)
+			page.Data = append(page.Data, o)
+		}
+		return IterateDone
+	}
+	oi = NewObjectIterator(ctx, nextFn, nil)
+	return
+}
+
+func (s *Storage) listFiles(ctx context.Context, path string) (fs []*drive.File, err error) {
+	var pathId string
+	pathId, err = s.pathToId(ctx, path)
+	if err != nil {
+		return fs, err
+	}
+	return s.listFilesInDir(ctx, pathId)
+}
+
+// List files in a folder by passing the folder's FileID
+func (s *Storage) listFilesInDir(ctx context.Context, fileId string) ([]*drive.File, error) {
+	searchArg := fmt.Sprintf("parents='%s'", fileId)
+	return s.rawListFiles(ctx, searchArg)
 }
 
 func (s *Storage) metadata(opt pairStorageMetadata) (meta *StorageMeta) {
-	panic("not implemented")
+	meta = NewStorageMeta()
+	meta.Name = s.name
+	meta.WorkDir = s.workDir
+	return meta
+}
+
+func (s *Storage) pathToId(ctx context.Context, path string) (fileId string, err error) {
+	absPath := s.getAbsPath(path)
+	pathUnits := strings.Split(absPath, "/")
+
+	for i := 0; i < len(pathUnits); i++ {
+		if i == 0 {
+			fileId, err = s.searchContentInDir(ctx, "root", pathUnits[i])
+		}
+		fileId, err = s.searchContentInDir(ctx, fileId, pathUnits[i])
+	}
+
+	if err != nil {
+		return "", err
+	}
+	return fileId, nil
+}
+
+func (s *Storage) rawListFiles(ctx context.Context, searchArg string) ([]*drive.File, error) {
+	var fs []*drive.File
+	pageToken := ""
+	for {
```

Comment:
If we need paging, please do them in our `NextObjectPage` function. Our object iterator will help to do the job.